src/tests/TestCryptographicRandom/TestCryptographicRandom.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2005-2017 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include <QtCore>
+#include <QtTest>
+
+#include "CryptographicRandom.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <limits>
+
+class TestCryptographicRandom : public QObject {
+		Q_OBJECT
+	private slots:
+		void fillBuffer();
+		void uint32();
+		void uniform();
+};
+
+// Verify the entropy of the data returned by CryptographicRandom::fillBuffer
+// by zlib compressing it and ensuring the compressed size is at least 99% of
+// the size of the input data.
+//
+// The inspiration for this test is from the rand_test.go file for Go standard
+// library's crypt/rand package:
+// https://golang.org/src/crypto/rand/rand_test.go
+void TestCryptographicRandom::fillBuffer() {
+	const int buflen = 1000000;
+
+	for (int i = 0; i < 10; i++) {
+		unsigned char *buf = reinterpret_cast<unsigned char *>(calloc(buflen, 1));
+
+		CryptographicRandom::fillBuffer(buf, buflen);
+
+		QByteArray flated = qCompress(buf, buflen);
+
+		QVERIFY(flated.size() >= ((buflen*99)/100));
+
+		free(buf);
+	}
+}
+
+void TestCryptographicRandom::uint32() {
+	for (int i = 0; i < 10000; i++) {
+		uint32_t val = CryptographicRandom::uint32();
+
+		// This is plea to the compiler to not optimize this test fully away.
+		uint64_t val64 = val;
+		QVERIFY(val64 <= std::numeric_limits<uint32_t>::max());
```

Comment:
Yes, you're right, the test is not very sensible... I wonder how we can test it.